### PR TITLE
Add fit_least_squares

### PR DESCRIPTION
**Pull request by moeyensj**
Adds a differential correction function that uses `scipy.optimize.least_squares` to perform a least squares fit of an orbit to observations. Also adds a function to remove the worst outlier, however, this PR does not add any logic or procedures to iteratively handle outlier rejection. This is best implemented by external users of these lower-level functions.

### CODE REVIEW

File: adam_core/orbit_determination/differential_correction.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+
+def fit_least_squares(
+    orbit: Orbits,
+    observations: OrbitDeterminationObservations,
+    propagator: Propagator,
+    **kwargs
+) -> Tuple[FittedOrbits, FittedOrbitMembers]:
+    """
+    Diffentially correct a fitted orbit using least squares.
```
Comment (akoumjian) [inline adam_core/orbit_determination/differential_correction.py:88]:
```suggestion
    Differentially correct a fitted orbit using least squares.
```

File: adam_core/orbit_determination/differential_correction.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    fitted_orbit : `~adam_core.orbit_determination.FittedOrbits` (1)
+        Fitted orbit.
+    fitted_orbit_members : `~adam_core.orbit_determination.FittedOrbitMembers` (N)
+        Fitted orbit members.
+    """
+    assert len(orbit) == 1, "Only one orbit can be differentially corrected"
+
+    # TODO: Investigate whether we want to add fitting for the epoch as well
+    # Set up least squares problem
+    func = residual_function
```
Comment (akoumjian) [inline adam_core/orbit_determination/differential_correction.py:119]: Why rename this if it's not parameterized?

**Review by akoumjian — APPROVED** [review]

File: adam_core/orbit_determination/differential_correction.py (hunk 2 above)
Comment (moeyensj) [reply adam_core/orbit_determination/differential_correction.py:119]: I was toying around with a second version of the function that fits the epoch as well. I'll remove this.

Comment (moeyensj) [reply adam_core/orbit_determination/differential_correction.py:119]: I was toying around with a second version of the function that fits the epoch as well. I'll remove this.

**Pull request merged**